e3db/types/queryadv.py

```diff
@@ -0,0 +1,150 @@
+
+        Parameters
+        """
+        self.__next_token = int(last_index)
+        self.__limit = int(count)
+        self.__include_data = bool(include_data)
+        self.__include_all_writers = bool(include_all_writers)
+        self.__match = match if match else [] 
+        self.__exclude = exclude if exclude else []
+        self.__range = None
+        
+    def append_match(self, p):
+        self.__match.append(p)
+
+    def append_exclude(self, p):
+        self.__exclude.append(p)
+
+    # count getters
+    @property
+    def count(self):
+        """
+        Get count of Query.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        int
+            Get count of records.
+        """
+        return self.__limit
+
+    # after_index getters
+    @property
+    def after_index(self):
+        """
+        Get after_index of Query.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        int
+            Get after_index of the Query.
+        """
+        return self.__next_token
+
+    @after_index.setter
+    def after_index(self, value):
+        """
+        Set after_index of QueryResult
+
+        Parameters
+        ----------
+        value: int
+            Set after_index from server result
+
+        Returns
+        -------
+        None
+        """
+        self.__next_token = int(value)
+
+    # include_data getters
+    @property
+    def include_data(self):
+        """
+        Get included_data of Query.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        bool
+            Whether data is included with the records.
+        """
+        return self.__include_data
+
+
+    # include_all_writers getters
+    @property
+    def include_all_writers(self):
+        """
+        Get whether the query included all writers.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        bool
+            include_all_writers
+        """
+        return self.__include_all_writers
+
+    def to_json(self):
+        """
+        Serialize the configuration as JSON-style object.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        dict
+            JSON-style document containing the Query elements.
+        """
+        return {
+            "limit": int(self.__limit),
+            "next_token": int(self.__next_token),
+            "include_all_writers": bool(self.__include_all_writers),
+            "include_data": bool(self.__include_data),
+            "match": [i.to_json() for i in self.__match],
+            "exclude": [i.to_json() for i in self.__exclude],
+            "range": self.__range.to_json() if self.__range is not None else {}
+        }
+
+    def match(self, cond="OR", strat="EXACT", writer=[], record=[], user=[], record_type=[], keys=[], values=[], plain=None):
```

Comment:
Mentioned this earlier but it should be noted somewhere what strategies and conditions are supported (or at least a link to another doc that highlights that.